### DISCUSSION TITLE
Benchmark vs Brotli

### DISCUSSION THREAD

**Discussion by RDIL** [kriszyp/msgpackr#88] [Ideas]
Hey all,

I'd be very curious to see a comparison to the encode/decode time as well as memory usage while doing both, to see which will be a better choice.
If anybody is willing to gather these statistics, that would be super awesome!

Thanks!

**Comment by kriszyp** [comment]
I'm curious what you are asking for. Brotli is a compression algorithm and MessagePack is a serialization format. Are you looking for a comparison of Brotli + JSON vs MessagePack (without compression)? You could also certainly use MessagePack with Brotli.

**Reply by RDIL** [reply]
MessagePack is a much smaller format than json, so I’m interested in MessagePack vs Brotli-compressed JSON.

**Reply by kriszyp** [reply]
Here is what I get with my sample data from the unit tests, using a Brotli compression level of 1 (higher levels produce a little smaller output, but take longer):

JSON.stringify + brotliCompressSync (461ms) - 2444 bytes compressed (JSON is 7569 uncompressed)
brotliDecompressSync + JSON.parse (436ms)
Plain MessagePack encoding: 6357 bytes
MessagePapck using msgpackr w/ record extension: 3821 bytes
unpack (85ms) (using record extension)
pack (237ms) (using record extension)

Obviously YMMV.

**Reply by RDIL** [reply]
Great!  Thank you!
